pkg/services/queryhistory/client.go

```diff
@@ -0,0 +1,103 @@
+package queryhistory
+
+import (
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+
+	authlib "github.com/grafana/authlib/types"
+	preferencesV1 "github.com/grafana/grafana/apps/preferences/pkg/apis/preferences/v1alpha1"
+	"github.com/grafana/grafana/pkg/apimachinery/identity"
+	"github.com/grafana/grafana/pkg/services/apiserver"
+	contextmodel "github.com/grafana/grafana/pkg/services/contexthandler/model"
+)
+
+type k8sClients struct {
+	namespacer     authlib.NamespaceFormatter
+	configProvider apiserver.DirectRestConfigProvider
+}
+
+// GetStars implements K8sClients.
+func (k *k8sClients) GetStars(c *contextmodel.ReqContext) ([]string, error) {
+	dyn, err := dynamic.NewForConfig(k.configProvider.GetDirectRestConfig(c))
+	if err != nil {
+		return nil, err
+	}
+	client := dyn.Resource(preferencesV1.StarsResourceInfo.GroupVersionResource()).Namespace(k.namespacer(c.OrgID))
+
+	ctx := c.Req.Context()
+	user, err := identity.GetRequester(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	obj, _ := client.Get(ctx, "user-"+user.GetIdentifier(), v1.GetOptions{})
+	if obj != nil {
```

Comment:
nit - this is pretty nested, might be better to return early?
```suggestion
	if obj == nil {
		return []string{}, nil
	}
```